deckhouse-controller/pkg/controller/module-controllers/release/controller.go

```diff
@@ -201,6 +202,8 @@ func (c *moduleReleaseReconciler) deleteReconcile(ctx context.Context, mr *v1alp
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	extenders.DeleteConstraints(mr.GetModuleName())
```

Comment:
I could be wrong, but it should've been placed under the `if mr.Status.Phase == v1alpha1.PhaseDeployed` condition, otherwise it would delete all constraints by the moduleName even if it was a reconcile for a Superseded/Suspended/Pending `ModuleRelease`.